protos/core/Btfs.proto

```diff
@@ -0,0 +1,230 @@
+syntax = "proto3";
+
+
+
+package btfs;
+
+
+
+
+
+// service EscrowService{. //FYI from Cong's review, this part will be removed eventually
+// 	submitContracts()
+// 	fullPay()
+// 	checkContract()
+// 	cancelContract()
+// }
+
+service GuardService{
+	rpc submitFileStoreMeta(FileStoreMeta) returns (Result){}
+	rpc health() returns(HealthStatus){} //used for other nodes check guard health , TBD input argument
+  	rpc sendQuestions(Questions) returns(Result){} //used for renter send the questions to guard
+  	rpc checkFileStoreMeta(CheckFileStoreMetaRequest) returns (FileStoreMeta){} //reserved for other nodes to check contract status
+}
+
+service HostService{
+	rpc challenge(ProofOfReplicateChallenge) returns (ProofOfReplicateChallenge){} //used for receieve challenge question and return the answer, TBD seems can be removed as the BTFS has embed function
+	rpc cancelContract(Contract) returns (Result){} 
+}
+
+//the relationship of the objects from Guard's view
+// FileStoreMeta (1:n) Contract (1:n) payout
+
+
+
+
+
+
+//contract is the signed document between the renter and host for one shard's storage
+// the contract information contains buyer, seller, escrow, guard , storage shard's information, rent duration, amount, payment method(periodically and times)
+message Contract{
+	bytes contract_id = 1; //one id which generated by renter, and will be check unique by escrow service
+	bytes renter_address = 2; //renter's wallet address
+	bytes host_address = 3; //host’s wallet address
+	bytes shard_hash = 4; //shard’s hash
+	int shard_index = 5; //shard’s sequence of the original file
+	int64 shard_file_size = 6; //shard’s file size
+	bytes file_hash = 7; //original file's hash, reserved point back to FileStoreMeta
+	Timestamp rent_start = 7; //rent start time
+	Timestamp rent_end = 8 ; //rent end time
+	bytes guard_address = 9; //renter proposed guard's address
+	bytes escrow_address = 10; //renter proposed escrow's address
+	bytes renter_propose_signature = 11; //renter signature for proposed information, signed for above information
+	int64 agreed_price = 12; //the price which agreed by host and rental, on the first phase, it equal to the proposed_price
+	int64 agreed_amount = 13; //the amount which agreed by host and rental, on the first phase, it should be equal file_size*duration*agreed_price
```

Comment:
It will always be `file_size*duration*agreed_price`, so is this really necessary?